src/modules/Telemetry/EnvironmentTelemetry.cpp

```diff
@@ -73,32 +75,31 @@ int32_t EnvironmentTelemetryModule::runOnce()
             DEBUG_MSG("Environment Telemetry: Initializing\n");
             // it's possible to have this module enabled, only for displaying values on the screen.
             // therefore, we should only enable the sensor loop if measurement is also enabled
+            
             switch (moduleConfig.telemetry.environment_sensor_type) {
-
-            case TelemetrySensorType_DHT11:
-            case TelemetrySensorType_DHT12:
-            case TelemetrySensorType_DHT21:
-            case TelemetrySensorType_DHT22:
-                return dhtSensor.runOnce();
-            case TelemetrySensorType_DS18B20:
-                return dallasSensor.runOnce();
-            case TelemetrySensorType_BME280:
-                return bme280Sensor.runOnce();
-            case TelemetrySensorType_BME680:
-                return bme680Sensor.runOnce();
-            case TelemetrySensorType_MCP9808:
-                return mcp9808Sensor.runOnce();
-            default:
-                DEBUG_MSG("Environment Telemetry: Invalid sensor type selected; Disabling module");
-                return (INT32_MAX);
+                case TelemetrySensorType_DHT11:
+                case TelemetrySensorType_DHT12:
+                case TelemetrySensorType_DHT21:
+                case TelemetrySensorType_DHT22:
+                    result = dhtSensor.runOnce();
+                case TelemetrySensorType_DS18B20:
+                    result = dallasSensor.runOnce();
+                default:
```

Comment:
Up till here you will always return dallasSensor. The old code "return"ed, e.g. jumped out of the switch, this one will just fall through with a new assignment. add a 'break' after dhtSensor.runOnce(); and dallasSensor.runOnce();

Reply:
Oh good catch. I got code blind and completely missed that when I was gutting that init method